src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -659,9 +664,20 @@ private void ReportAssignmentOperatorError(AssignmentExpressionSyntax node, Diag
 
         private static void ReportBinaryOperatorError(ExpressionSyntax node, DiagnosticBag diagnostics, SyntaxToken operatorToken, BoundExpression left, BoundExpression right, LookupResultKind resultKind)
         {
+            if (left.Kind == BoundKind.SuppressNullableWarningExpression)
+            {
+                Error(diagnostics, ErrorCode.ERR_IllegalSuppression, left.Syntax);
+                return;
+            }
```

Comment:
This pattern is repeated a number of places, including in `errorIfSuppression` local function. Consider extracting a top-level method.